routes/login/login.session.spec.js

```diff
@@ -0,0 +1,197 @@
+const session = require('supertest-session')
+const { extractCsrfToken, withCSRF } = require('../utils.spec')
+const cheerio = require('cheerio')
+const app = require('../../app.js')
+
+let csrfToken, cookie, testSession
+
+const doAccessCode = (code = 'A5G98S4K1') => {
+  return request => {
+    request.use(withCSRF(cookie, csrfToken)).send({
+      code,
+      redirect: '/login/sin',
+    })
+  }
+}
+
+const doSIN = (sin = '847 339 283') => {
+  return request => {
+    request.use(withCSRF(cookie, csrfToken)).send({
+      sin,
+      redirect: '/login/dateOfBirth',
+    })
+  }
+}
+
+const doDateofBirth = ({ dobDay = '09', dobMonth = '09', dobYear = '1977' } = {}) => {
+  return request => {
+    request.use(withCSRF(cookie, csrfToken)).send({
+      dobDay,
+      dobMonth,
+      dobYear,
+      redirect: '/personal/name',
+    })
+  }
+}
+
+describe('Test /login SESSION responses', () => {
+  beforeEach(async () => {
+    testSession = session(app)
+    const getresp = await testSession.get('/login/code')
+    cookie = getresp.headers['set-cookie']
+    csrfToken = extractCsrfToken(getresp)
+  })
+
+  // TEST DONT RETURN TO ACCESS CODE PAGE IF ONLY SIN (NO ACCESS CODE OR DOB)
+  test('it returns a 200 on /login/dob page when only the SIN exists', async () => {
```

Comment:
We might be able to reduce some of the code here by using couple maps for requests that do and don't lead to various places,  like we do here in login.spec:

```badDoBRequests.map(badRequest => {
        test(`it returns a 422 with: "${badRequest.label}"`, async () => {
          const response = await authSession
            .post('/login/dateOfBirth')
            .use(withCSRF(cookie, csrfToken))
            .send({ ...badRequest.send })
          expect(response.statusCode).toBe(422)
        })
      })```

Reply:
We could do, but generally the more abstract the tests are, the harder they are to read. I was talking with @nmakuch a few days ago about some of the tests that were being mapped to explain how they worked.

For these ones, I am kind of okay with the copy+paste but if we wanted to make it less lines, I think checking the responses (for errors, etc) would save a bunch of copy + paste.

Reply:
Fair enough, they made sense to me but I can see how they'd be abstract enough to be not worth the code savings. 👍